### PR TITLE
Pages Tab: Favorites section

### LINKED ISSUE #5251: Favorites section in the Pages tab
Image

### PR DESCRIPTION
**Pull request by balazsbajorics**
image
image

This PR adds a Favorites section to the Pages tab.

If the user navigates to a route which is not yet a favorite, a "placeholder" favorite appears, clicking on the star adds it to the list of favorites in package.json. 
The already-favorited routes can be unfavorited by clicking on the star again.

The Routes section shows a tree view of all favorited routes. (A Follow Up PR will make it so that Routes shows -all- possible remix Routes)

Fixes #5251

### CODE REVIEW

File: editor/src/components/canvas/remix/remix-utils.spec.ts [hunk 1]
```diff
@@ -45,64 +45,111 @@ describe('addNewFeaturedRouteToPackageJson', () => {
...
+  it('errors if the featured routes prop if not an array', async () => {
+    expect(() =>
+      addNewFeaturedRouteToPackageJson('foo.bar')(
+        `{ "hey": "there", "utopia": {"featuredRoutes": "WRONG"} }`,
+      ),
+    ).toThrow('should be an array')
+  })
+})
+
+describe('removeFeaturedRouteFromPackageJson', () => {
```
Comment (balazsbajorics) [inline editor/src/components/canvas/remix/remix-utils.spec.ts:95]: I generated these tests using Copilot

File: editor/src/components/navigator/left-pane/pages-pane.tsx [hunk 2]
```diff
@@ -282,12 +321,146 @@ const PageRouteEntry = React.memo<PageRouteEntryProps>((props) => {
...
+        dispatch([addNewFeaturedRoute(url)])
+      } else {
+        dispatch([removeFeaturedRoute(url)])
+      }
+      e.stopPropagation()
+    },
+    [dispatch, url, addedToFavorites],
+  )
+
+  const [mouseOver, setMouseOver] = React.useState(false)
```
Comment (balazsbajorics) [inline editor/src/components/navigator/left-pane/pages-pane.tsx:415]: I tried to not rely on mouseOver events,and instead have two Icn components below each other, but I gave up after half an hour of struggling.

File: editor/src/components/navigator/left-pane/pages-pane.tsx [hunk 3]
```diff
@@ -128,28 +136,40 @@ export const PagesPane = React.memo((props) => {
...
+  const activeRouteDoesntMatchAnyFavorites = !featuredRoutes.includes(activeRoute)
+
   return (
     <FlexColumn style={{ height: '100%', overflowY: 'scroll' }}>
       <InspectorSectionHeader>
-        <FlexRow style={{ flexGrow: 1 }}>Pages</FlexRow>
+        <FlexRow style={{ flexGrow: 1 }}>Favorites</FlexRow>
+      </InspectorSectionHeader>
+      <FlexColumn style={{ paddingBottom: 24 }}>
+        {featuredRoutes.map((favorite: string) => {
```
Comment (ruggi) [inline editor/src/components/navigator/left-pane/pages-pane.tsx:148]: not for this PR, but we could/should be aggressive and make sure you don't have any duplicates in the favorite routes (for the same Remix path) so we avoid having a bunch of issues with duplicate keys

**Review by ruggi — APPROVED** [review]

File: editor/src/components/navigator/left-pane/pages-pane.tsx (hunk 3 above)
Comment (balazsbajorics) [reply editor/src/components/navigator/left-pane/pages-pane.tsx:148]: it's a feature to have featured routes showcasing the same remix path with different variables!

File: editor/src/components/navigator/left-pane/pages-pane.tsx [hunk 4]
```diff
@@ -82,7 +90,7 @@ export const PagesPane = React.memo((props) => {
             }),
           )
         }),
-        (l, r) => l?.resolvedPath === r?.resolvedPath,
+        (l, r) => l?.path === r?.path,
```
Comment (balazsbajorics) [inline editor/src/components/navigator/left-pane/pages-pane.tsx:93]: @ruggi this should mitigate the duplication of Featured URL -> Remix path

**Review by liady — APPROVED** [review]

**Pull request merged**